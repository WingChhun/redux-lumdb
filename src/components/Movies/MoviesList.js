/* eslint react/no-did-mount-set-state: 0 */
import React, {PureComponent} from 'react';
import styled from 'styled-components';
import Movie from './Movie';

class MoviesList extends PureComponent {

    constructor(props)
    {
        super(props);

        this.state = {
            movies: []
        }
    }

    componentWillMount = () => {
        this
            .props
            .getMovies();
    }

    componentWillReceiveProps = props => {
        this.setState({movies: props.movies})

    }

    render() {

        const {movies} = this.state;

        return (
            <MovieGrid>
                {movies && movies.map(movie => <Movie key={movie.id} movie={movie}/>)}
            </MovieGrid>
        );
    }
}

export default MoviesList;

const MovieGrid = styled.div `
  display: grid;
  padding: 1rem;
  grid-template-columns: repeat(6, 1fr);
  grid-row-gap: 1rem;
`;
