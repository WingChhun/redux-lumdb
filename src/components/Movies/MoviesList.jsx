/* eslint react/no-did-mount-set-state: 0 */
import React, {PureComponent} from 'react';
import styled from 'styled-components';
import Movie from './Movie';
import PropTypes from 'prop-types';
class MoviesList extends PureComponent {

    constructor(props)
    {
        super(props);

        this.state = {
            movies: props.movies
        }
    }

    componentDidMount = () => {

        const {isLoaded, getMovies, moviesLoadedAt} = this.props;
        const oneHour = 60 * 60 * 1000;
        if (!isLoaded || (new Date() - moviesLoadedAt) > oneHour) {
            getMovies();
        }
    }

    componentWillReceiveProps = props => {
        this.setState({movies: props.movies})
    }

    render() {

        const {movies} = this.state;

        return (
            <MovieGrid>
                {movies && movies.map(movie => <Movie key={movie.id} movie={movie}/>)}
            </MovieGrid>
        );
    }
}

export default MoviesList;

const MovieGrid = styled.div `
  display: grid;
  padding: 1rem;
  grid-template-columns: repeat(6, 1fr);
  grid-row-gap: 1rem;
`;

MoviesList.propTypes = {

    movies: PropTypes.array.isRequired,
    getMovies: PropTypes.func.isRequired

}