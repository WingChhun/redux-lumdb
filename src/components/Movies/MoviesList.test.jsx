import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import MoviesList from "./MoviesList.jsx";

Enzyme.configure({adapter: new Adapter()})

describe("MoviesList", () => {

    const props = {

        movies: [
            {
                id: 1,
                poster_path: 'hello',
                title: "Yo"
            }, {
                id: 2,
                poster_path: 'hello',
                title: "Yo"
            }, {
                id: 3,
                poster_path: 'hello',
                title: "Yo"
            }
        ],
        getMovies: jest.fn()

    }

    it("Renders properly if given movies", () => {
        const wrapper = shallow(<MoviesList {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it("componentWillReceiveProps shoul;d update movies State", () => {

        const newProps = {
            ...props,
            movies: [
                {
                    id: 2,
                    title: "dude",
                    poster_path: '/path'
                }
            ]
        }
        const wrapper = shallow(<MoviesList {...props}/>);
        const instance = wrapper.instance();

        instance.componentWillReceiveProps(newProps);
    
        expect(instance.state.movies).toBe(newProps.movies);
    });



});