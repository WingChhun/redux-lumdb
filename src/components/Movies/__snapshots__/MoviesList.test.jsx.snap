// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList Renders properly if given movies 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesList
    getMovies={[Function]}
    movies={
        Array [
            Object {
              "id": 1,
              "poster_path": "hello",
              "title": "Yo",
            },
            Object {
              "id": 2,
              "poster_path": "hello",
              "title": "Yo",
            },
            Object {
              "id": 3,
              "poster_path": "hello",
              "title": "Yo",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Movie
          movie={
                    Object {
                              "id": 1,
                              "poster_path": "hello",
                              "title": "Yo",
                            }
          }
/>,
        <Movie
          movie={
                    Object {
                              "id": 2,
                              "poster_path": "hello",
                              "title": "Yo",
                            }
          }
/>,
        <Movie
          movie={
                    Object {
                              "id": 3,
                              "poster_path": "hello",
                              "title": "Yo",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "movie": Object {
            "id": 1,
            "poster_path": "hello",
            "title": "Yo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "movie": Object {
            "id": 2,
            "poster_path": "hello",
            "title": "Yo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "movie": Object {
            "id": 3,
            "poster_path": "hello",
            "title": "Yo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Movie
            movie={
                        Object {
                                    "id": 1,
                                    "poster_path": "hello",
                                    "title": "Yo",
                                  }
            }
/>,
          <Movie
            movie={
                        Object {
                                    "id": 2,
                                    "poster_path": "hello",
                                    "title": "Yo",
                                  }
            }
/>,
          <Movie
            movie={
                        Object {
                                    "id": 3,
                                    "poster_path": "hello",
                                    "title": "Yo",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "movie": Object {
              "id": 1,
              "poster_path": "hello",
              "title": "Yo",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "movie": Object {
              "id": 2,
              "poster_path": "hello",
              "title": "Yo",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "movie": Object {
              "id": 3,
              "poster_path": "hello",
              "title": "Yo",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
